### DISCUSSION TITLE
Route name readability

### DISCUSSION THREAD

**Discussion by ManojKiranA** [filamentphp/filament#3686] [Ideas]
After this [tweet](https://twitter.com/ProSaad5/status/1562845159389220864) while checking the code what may be issue.

I have my own base resource class which generates the `slug` by combining the navigation group

```
<?php

namespace App\Filament\Base\Resources;

use Filament\Resources\Resource;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

class BaseResource extends Resource
{
    public static function getGlobalSearchResultUrl(Model $record): string
    {
        return static::getUrl('view', ['record' => $record]);
    }

    public static function getSlug(): string
    {
        return collect([
            static::getNavigationGroup(),
            ...Str::of(parent::getSlug())->explode('/'),
            ])
            ->map(fn ($each) => Str::of($each)->lower()->slug('-')->toString())
            ->implode('/');
    }
}

```

So the `getRoutes` method on [Resource](https://github.com/filamentphp/filament/blob/75ceecd527fb933d4b72b676e82e92653b0e6010/packages/admin/src/Resources/Resource.php#L336) page is using the slug to generate the route name It adds `/` in the route name.

Something similar to this

![image](https://user-images.githubusercontent.com/30294553/186891722-132f3a5f-505e-4e6a-a49e-5427e20cc226.png)

I that case we can add one more method to determine the route name. But this is a breaking change if someone hardcoded the routes instead of using `Resource::getUrl('view', ['record' => $record]);`

If you are ok with using dot separated route let me know.

**Comment by danharrin** [comment]
Hi, this would be a breaking change. People are welcome to put the navigation group in their `$slug` if they want.

**Comment by ManojKiranA** [comment]
The only problem I see is using the getSlug method to generate route name
